server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -386,7 +387,7 @@ public void testShouldBeYellowWhenRestartingReplicasReachedAllocationDelay() {
                             List.of(ImpactArea.SEARCH)
                         )
                     ),
-                    List.of(new Diagnosis(ACTION_CHECK_ALLOCATION_EXPLAIN_API, List.of("restarting-index")))
+                    List.of(new Diagnosis(DIAGNOSIS_WAIT_FOR_OR_FIX_DELAYED_SHARDS, List.of("restarting-index")))
```

Comment:
I'm a little hazy on how the shutdown delays work. From my understanding the DELAYED_ALLOCATION status can only be reached from node shutdown logic. Is this status something that requires intervention to clear or will it still be cleared like the other delayed allocation situations?

Reply:
Not sure I understand your question. What would are the other delayed allocation situations we would encounter?

Reply:
I'm not sure about what other delayed allocation situations we would encounter. I couldn't find any others except for the node shutdown.

My main question is whether this status is something that shards can get stuck in and thus need intervention to fix, or if once the delay has elapsed will the shard transition to another stage of recovery which may or may not succeed for other reasons. Based on what I'm reading, it seems like the answer is the latter of the two options, which would be great if that is the case.

Reply:
My understanding is that the latter is the case. Could we add an integration test for this? (ie. lapse the delay and get a different diagnosis for that same shard?)

Reply:
I'm not sure how I would go about building that integration test out unfortunately. If the system is designed to automatically attempt allocation after the delay, then validating that behavior seems like something out of scope for the health indicator to attempt simulating.

Reply:
Unless we have a specific and reproducible edge case for this, I think we'd be ok with relying on the underlying system's invariants for now.